daml-lf/spec/daml-lf-1.rst

```diff
@@ -1327,6 +1283,120 @@ Then we define *well-formed expressions*. ::
   explicit syntax for maps being forbidden in serialized programs.
 
 
+Well-formed case alternatives
+.............................
+
+Case expressions ``Γ  ⊢  'case' e 'of' alt₁ | … | altₙ : σ`` require the
+notion of well-formed case alternatives ``Γ ⊢ τ // alt : σ``  defined here.
+To simplify the presentation, we omit the assumption that the scrutinee
+type ``τ`` is well-formed, in the rules below. ::
+
+                                ┌──────────────────┐
+  Well-formed case alternatives │ Γ ⊢ τ // alt : σ │
+                                └──────────────────┘
+
+      'variant' T (α₁:k₁) … (αₙ:kₙ) ↦ … | Vᵢ : σᵢ | …  ∈  〚Ξ〛Mod
+      σᵢ  ↠  σᵢ'
+      x : σᵢ'[α₁ ↦ τ₁, …, αₙ ↦ τₙ] · Γ  ⊢  e : σ
+    ——————————————————————————————————————————————————————————————— AltVariant
+      Γ  ⊢  Mod:T τ₁ … τₙ  //  Mod:T:V x  →  e : σ
```

Comment:
```suggestion
      Γ  ⊢  Mod:T τ₁ … τₙ  //  Mod:T:Vᵢ x  →  e : σ
```
or drop the `i` indices above, which probably makes it easier to read.

Reply:
Good point!

Reply:
I dropped the i indices and renamed σᵢ to τ and σᵢ' to τ', reserving σ only for the output type.